### PR TITLE
Allow git to open notebook files

### PR DESCRIPTION
**Pull request by chlafreniere**
Fix for #7385. Note that actually doing the diff is _not_ in scope for right now 😄. Just not failing is the first priority here 😄 

Example:
image

### CODE REVIEW

File: src/sql/workbench/parts/notebook/browser/models/notebookInput.ts [hunk 1]
```diff
@@ -38,7 +39,7 @@ export class NotebookEditorModel extends EditorModel {
 	private readonly _onDidChangeDirty: Emitter<void> = this._register(new Emitter<void>());
 	private _lastEditFullReplacement: boolean;
 	constructor(public readonly notebookUri: URI,
-		private textEditorModel: TextFileEditorModel | UntitledEditorModel,
+		private textEditorModel: TextFileEditorModel | UntitledEditorModel | ResourceEditorModel,
```
Comment (chlafreniere) [inline src/sql/workbench/parts/notebook/browser/models/notebookInput.ts]: Any thoughts on how I can make this cleaner throughout? Can I reference all of these as a type or something?

Comment (alanrenmsft) [reply src/sql/workbench/parts/notebook/browser/models/notebookInput.ts]: yes, you can do this. export type TextEditorModel = TextFileEditorModel | UntitledEditorModel | ResourceEditorModel;

Comment (anthonydresser) [reply src/sql/workbench/parts/notebook/browser/models/notebookInput.ts]: You could just have the type be `BaseTextEditorModel`

Comment (anthonydresser) [reply src/sql/workbench/parts/notebook/browser/models/notebookInput.ts]: From what I can tell text file, untitled and resource editor are the only classes that extend it.

Comment (chlafreniere) [reply src/sql/workbench/parts/notebook/browser/models/notebookInput.ts]: Thanks. I cleaned this up everywhere else (except where the logic got really hairy... like right here. I'd rather leave this for now 😄)

**Pull request merged**